VERSIONING.md

```diff
@@ -0,0 +1,94 @@
+# Versioning
+
+## SemVer and shorter SemVer
+
+We use [semantic versioning] for all of our images. The images with fully
+specified version (e.g. `vX.Y.Z`) are immutable and will never be changed. We
+also support a shorter version of semantic versioning. E.g.
+`v<major> and v<major>.<minor>`.
+
+The shorter version of semantic versioning is going to be "moving tags":
+
+- `v<major>.<minor>` always points to `v<major>.<minor>.X` where X is the latest
+  (largest) patch version number. E.g. assume `v1.2` points `v1.2.0` initially.
+  After `v1.2.1` is released, `v1.2` now points to `v1.2.1`.
+  
+- `v<major>` always points to `v<major>.X` where X is the latest (largest) minor
+  version number. E.g. assume `v1` points `v1.2.0` initially. After `v1.3.0` is
+  released, `v1` now points to `v1.3.0`. After `v1.3.1` is released, `v1` now
+  points to `v1.3.1`.
+
+Note: We do NOT support the `latest` tag, since we cannot provide any
+compatibility guarantee for it, and the pipeline won’t produce deterministic
+results.
+
+## Interfaces
+
+All functions in the catalog comply with the [function spec]. To learn more
+about the functions concept, see [here](http://kpt.dev/book/02-concepts/03-functions).
+
+There are 2 interfaces for a function:
+- The `functionConfig`
+- The function itself
```

Comment:
How's "the function" an interface. Rephrase to be more precise.

Reply:
Now we avoiding using "API" and "interface" here. I can't come up with a good word here. Can you please suggest the wording here?

Reply:
@frankfarzan @droot Any suggested wording here?

Reply:
We can use `Function behavior` term if it makes more sense.